Kickstarter-iOS/Views/Cells/RewardCell.swift

```diff
@@ -2,32 +2,352 @@ import Foundation
+    self.containerView.addGestureRecognizer(tapGestureRecognizer)
   }
 
-  func configureWith(value _: Reward) {}
+  private func setupConstraints() {
+    let baseStackView = self.baseStackView
+    let containerView = self.containerView
+    let pledgeButton = self.pledgeButton
+    let pledgeButtonLayoutGuide = self.pledgeButtonLayoutGuide
+
+    // Container view
+    NSLayoutConstraint.activate([containerView.widthAnchor.constraint(equalTo: self.contentView.widthAnchor)])
+
+    // Base stack view
+    _ = baseStackView
+      |> \.translatesAutoresizingMaskIntoConstraints .~ false
+
+    let containerMargins = containerView.layoutMarginsGuide
+
+    NSLayoutConstraint.activate([
+      baseStackView.leftAnchor.constraint(equalTo: containerMargins.leftAnchor),
+      baseStackView.rightAnchor.constraint(equalTo: containerMargins.rightAnchor),
+      baseStackView.topAnchor.constraint(equalTo: containerMargins.topAnchor)
+    ])
+
+    // Pledge button
+    _ = pledgeButton
+      |> \.translatesAutoresizingMaskIntoConstraints .~ false
+
+    let topConstraint = pledgeButton.topAnchor.constraint(equalTo: pledgeButtonLayoutGuide.topAnchor)
+    _ = topConstraint
+      |> \.priority .~ .defaultLow
+      |> \.isActive .~ true
+
+    let contentMargins = self.contentView.layoutMarginsGuide
+
+    NSLayoutConstraint.activate([
+      pledgeButton.leftAnchor.constraint(equalTo: contentMargins.leftAnchor),
+      pledgeButton.rightAnchor.constraint(equalTo: contentMargins.rightAnchor),
+      // swiftlint:disable:next line_length
+      pledgeButton.bottomAnchor.constraint(lessThanOrEqualTo: contentMargins.bottomAnchor),
+      // swiftlint:disable:next line_length
+      pledgeButton.heightAnchor.constraint(greaterThanOrEqualToConstant: Styles.minTouchSize.height)
+    ])
+
+    // Pledge button layout anchor
+    NSLayoutConstraint.activate([
+      pledgeButtonLayoutGuide.bottomAnchor.constraint(equalTo: containerMargins.bottomAnchor),
+      pledgeButtonLayoutGuide.leftAnchor.constraint(equalTo: containerMargins.leftAnchor),
+      pledgeButtonLayoutGuide.rightAnchor.constraint(equalTo: containerMargins.rightAnchor),
+      // swiftlint:disable:next line_length
+      pledgeButtonLayoutGuide.topAnchor.constraint(equalTo: baseStackView.bottomAnchor, constant: Styles.grid(3)),
+      pledgeButtonLayoutGuide.heightAnchor.constraint(equalTo: pledgeButton.heightAnchor)
+    ])
+  }
+
+  fileprivate func load(items: [String]) {
+    self.includedItemsStackView.subviews.forEach { $0.removeFromSuperview() }
+
+    let includedItemViews = items.enumerated().map { (index, item) -> [UIView] in
+      let label = UILabel()
+        |> baseRewardLabelStyle
+        |> sectionBodyLabelStyle
+        |> \.text .~ item
+
+      let separator = UIView()
+        |> separatorStyle
+      separator.heightAnchor.constraint(equalToConstant: 1).isActive = true
+
+      var itemViews: [UIView] = [label]
+
+      if index != items.endIndex - 1 {
```

Comment:
I tend to prefer `dropLast()` and `dropFirst()` over index comparisons 🤷‍♂ Here's a way one could do that but feel free to ignore if you feel it adds no material benefit or readability gain:

```swift
// we want all of the labels formatted the same
let includedItemViews = items.map { item -> UIView in
  let label = UILabel()
    |> baseRewardLabelStyle
    |> sectionBodyLabelStyle
    |> \.text .~ item

  return label
}

// we want all of them separated except the last one - dropLast()
let separatedItemViews = includedItemViews.dropLast().map { view -> [UIView] in
  let separator = UIView()
    |> separatorStyle
  separator.heightAnchor.constraint(equalToConstant: 1).isActive = true

  return [view, separator]
}
.flatMap { $0 }

let allItemViews = [self.includedItemsTitleLabel]
  + separatedItemViews
  + [includedItemViews.last].compact() // we append the last unseparated label here
```